clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -20,13 +20,20 @@
 public class MockTime implements Time {
 
     private long nanos = 0;
+    private long autoTickMs = 0;
 
     public MockTime() {
         this.nanos = System.nanoTime();
     }
 
+    public MockTime(long autoTickMs) {
+        this.nanos = System.nanoTime();
+        this.autoTickMs = autoTickMs;
```

Comment:
After a second thought: do we want to advance the same step in both milliseconds() and nanoseconds()? maybe we should only advance by TimeUnit.NANOSECONDS.convert(autoTickMs, TimeUnit.NANOSECONDS)?